### ISSUE TITLE
Does CachedConvTranspose1d use the right cache size?

### ISSUE THREAD

**Issue by leoauri** [acids-ircam/cached_conv#11]
At first glance, the CachedConvTranspose1d implementation looks a bit strange, and it throws errors for me when dropping in.

From [PyTorch nn.ConvTranspose1d docs](https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose1d.html):

> The `padding` argument effectively adds `dilation * (kernel_size - 1) - padding` amount of zero padding to both sizes of the input.

But the implementation looks like it uses `nn.ConvTranspose1d().padding[0]` to tell it how much padding to expect...

Am I missing something here? It's meant to work as a drop-in replacement for `nn.ConvTranspose1d`, right?
